### DISCUSSION TITLE
ucrtbase.dll Windows 10 error due to rtaudio.dll conflict

### DISCUSSION THREAD

**Discussion by WB0OCV** [AlexandreRouma/SDRPlusPlus#638] [General]
I've had sdrpp working on Windows 10 for several months.  Yesterday, 08-Feb-2022 I tried running it and it shut down immediately after starting.  The error log showed a problem with the ucrtbase.dll.  I tried removing and reinstalling sdrpp, reinstalled all of the Windows Visual C++ vcredist run-times (sometimes related to ucrtbase.dll), and even recompiled and reinstalled RtAudio support in Windows Program Files.  Nothing worked.  

I noticed when I deleted the rtaudio.dll file from the sdrpp folder, the program would function correctly, albeit without an audio sink.  I eventually checked the Windows Sound Control panel, and changed the default device for Windows audio playback.  After doing that, sdrpp would work fine with audio sink selectable.  It appears there was some conflict occurring between the sdrpp audio sink and the Windows default playback device that resulted in the ucrtbase error.  If you have any virtual audio cable programs installed, try changing the default Windows playback device to see if it rectifies your problem.

**Comment by AlexandreRouma** [comment]
Would be interesting to report that issue to the rtaudio developpers.
Could you elaborate on the state of the default device before and after the fix?

**Comment by WB0OCV** [comment]
Good point, Alex.  In order to pipe audio from the various SDR programs I use into other digital decoding programs (fldigi, MultiPSK, etc.), I installed VB-Audio Virtual Cable.  In the case of the sdrpp error with rtaudio, I had the Windows default playback device set to the PC Speakers, essentially bypassing the virtual cable pipe.  Once I set the default playback device to the virtual cable, sdrpp worked fine.  If I then selected the PC Speaker as the audio sink within sdrpp, the program would shut down and generate the error.  In the original case of having the default playback set to PC Speakers, sdrpp would start and immediately shut down.  If I deleted the rtaudio.dll so that sdrpp couldn’t load it, the program would work fine – I could operate the SDR (Airspy HF+, SDRPlay RSP1, RTL-SDR), see signals on the waterfall, change frequencies, modes, select different sinks, etc.  However, without the rtaudio.dll, audio did not show up as one of the possible sinks. Perhaps the error involves the interplay between sdrpp -> rtaudio.dll -> VAC -> Windows Speaker Control.  It was interesting to note that having the default playback set to PC Speakers did not affect the functionality of other SDR programs like SDRSharp, SDR Console, or HDSDR.  It only seemed to be an issue with sdrpp. Thanks for the inquiry, Alex, and thanks for making sdrpp available to us.  Have a wonderful day. Ken --  WBØOCV From: AlexandreRoumaSent: Wednesday, February 9, 2022 08:03 AMTo: AlexandreRouma/SDRPlusPlusCc: WB0OCV; AuthorSubject: Re: [AlexandreRouma/SDRPlusPlus] ucrtbase.dll Windows 10 error due to rtaudio.dll conflict (Discussion #638) Would be interesting to report that issue to the rtaudio developpers.Could you elaborate on the state of the default device before and after the fix?—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you authored the thread.Message ID: ***@***.***>

**Reply by AlexandreRouma** [reply]
that's very strange, having the speakers as default device shouldn't cause an issue. I wonder what the difference between your PC and the working PCs is. If I were able to replicate the right conditions I could properly identify where the issue happens.
